gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -103,21 +112,39 @@ private Set<GroupPartOrPackageName> fetchFoldersFromContent(String contents) {
     }
 
     private Set<DependencyVersion> parseVersionsFromContent(String content) {
-        Set<DependencyVersion> versions = new HashSet<>();
         try {
             XmlMapper xmlMapper = new XmlMapper();
+            xmlMapper.registerModule(new GuavaModule());
 
             Metadata metadata = xmlMapper.readValue(content, Metadata.class);
-            if (metadata.versioning() != null && metadata.versioning().versions() != null) {
-                String latest = metadata.versioning().latest();
-                for (String version : metadata.versioning().versions()) {
-                    versions.add(DependencyVersion.of(version, latest.equals(version)));
-                }
+            List<String> allVersions = new ArrayList<>(metadata.versioning().versions());
+            Collections.reverse(allVersions);
+
+            if (allVersions.isEmpty()) {
+                return Collections.emptySet();
             }
+
+            String latest = Optional.ofNullable(metadata.versioning().release())
+                    .filter(l -> !l.isEmpty())
+                    .orElseGet(() -> metadata.versioning().latest());
+
+            if (VERSION_PATTERN.matcher(latest.toLowerCase(Locale.ROOT)).matches()) {
```

Comment:
I also don't understand the predicate:

> if `latest` is a well formed version, we find the first non-well formed version and use that as the latest?

Reply:
Ah this is probably badly worded from me, the matcher is checking for verisons that are not good e.g. SNAPSHOT of beta versions, so if we latest is a snapshot version work backwards till we find the first version that is not a snapshot/beta/etc

Reply:
I've tried to tidy this up so hopefully it is a lot more self explanatory now